api/envoy/type/tracing/v2/custom_tag.proto

```diff
@@ -0,0 +1,91 @@
+syntax = "proto3";
+
+package envoy.type.tracing.v2;
+
+option java_outer_classname = "CustomTagProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.type.tracing.v2";
+
+import "envoy/type/metadata/v2/metadata.proto";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Custom Tag]
+
+// Describes custom tags for the active span.
+// [#next-free-field: 7]
+message CustomTag {
+  // Literal type custom tag with static value for the tag value.
+  message Literal {
+    // Static literal value to populate the tag value.
+    string value = 1 [(validate.rules).string = {min_bytes: 1}];
+  }
+
+  // Environment type custom tag with environment name and default value.
+  message Environment {
+    // Environment variable name to obtain the value to populate the tag value.
+    string name = 1 [(validate.rules).string = {min_bytes: 1}];
+
+    // When the environment variable is not found,
+    // the tag value will be populated with this default value if specified,
+    // otherwise no tag will be populated.
+    string default_value = 2;
+  }
+
+  // Header type custom tag with header name and default value.
+  message Header {
+    // Request header name to obtain the value to populate the tag value.
+    string name = 1 [(validate.rules).string = {min_bytes: 1}];
+
+    // When the request header does not exist,
```

Comment:
remove "request" as in the future you might add response headers, trailers, etc. below.